openedx/core/djangoapps/site_configuration/helpers.py

```diff
@@ -27,6 +22,25 @@ def get_current_site_configuration():
         return None
 
 
+def get_current_site_configuration():
+    """
+    Return configuration for the current site.
+
+    Returns:
+         (openedx.core.djangoapps.site_configuration.models.SiteConfiguration): SiteConfiguration instance associated
+         with the current site.
+    """
+    dummy_key = "singleton"
+    cache = RequestCache("site_config")
+    response = cache.get_cached_response(dummy_key)
+    if response.is_found:
+        return response.value
+    else:
+        config = _uncached_get_current_site_configuration()
+        cache.set(dummy_key, config)
+        return config
+
```

Comment:
There's a lot of repetition in the various uses of RequestCache, and I'm tempted to add the following method to it:

```python
    def get_or_compute(self, key, compute_value):
        """
        Retrieves a bare cached value (not a CachedResponse) for the provided
        key, or computes it and sets the value first on cache miss.

        Args:
            key (string)
            compute_value (function): Thunk that returns a value to cache and return
        """
        response = self.get_cached_response(key)
        if response.is_found:
            return response.value
        else:
            config = compute_value()
            self.set(key, config)
            return config
```

Then this method could be:

```python
def get_current_site_configuration():
    dummy_key = "singleton"
    return RequestCache("site_config").get_or_compute(dummy_key, _uncached_get_current_site_configuration)
```